lib/rust/ensogl/core/src/profiler.rs

```diff
@@ -0,0 +1,28 @@
+//! Enso specific metadata logging utilities for use with the profiling framework.
+use enso_profiler;
+use serde;
+use serde::Serializer;
+use std::fmt::Display;
+use std::fmt::Formatter;
+
+
+/// Metadata that is logged within the Enso core libraries.
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize, Eq, PartialEq)]
+pub enum Metadata {
+    /// An RPC event that was received from the backend.
+    RpcEvent(String),
```

Comment:
Why should an `ensogl` module know anything about RPC? `profiler` supports arbitrarily-many dependency-injected metadata types so that we can separate concerns--a metadata type defined here should know only about EnsoGL metadata, and we can have one somewhere else that's appropriate for RPC messages.

Reply:
To clarify this, the idea of this API design is that we can use a big enum that depends on everything when we're analyzing the data (because data consumers will need access to data type definitions from all over the app), but we shouldn't try to use such a cross-cutting enum when logging metadata from all over the codebase (it mixes concerns, and the dependencies all go circular). That's why `MetadataLogger` doesn't rely on an enum of all metadata types; instead it supports defining different variants of the enum at different places in the code. They come together in the serialized log, and then we can use a big all-knowing enum only for deserializing.

We will need a place to define the all-inclusive `Metadata` enum--a crate for Enso-specific profile-interpreting code, that is very high level; it will have dependencies on all the app's runtime crates that define metadata types. We could call it something like `enso-profiler-data-tools` (it defines metadata and tools for building on `enso-profiler-data`). When I implement the tool that integrates data from the backend I'll merge that into the same crate.

Reply:
We could gather the names all in one place that defines them as `const`s for use when registering each `MetadataLogger`--serialization and deserialization would still be decoupled, but at least it would be easy to compare the metadata `enum` with the list of names.

Reply:
That place could just be the independent crate, which would be imported in many places. But that crate would only need some 3rd party libs and the profiling crate. So, no circular dependencies.  

I think I like that better.

Reply:
But we also need those structs for deserialisaiton anyway.